experimental/examples/bubblebench/editable-shared-tree/src/client.ts

```diff
@@ -3,14 +3,12 @@
  * Licensed under the MIT License.
  */
 import { IClient } from "@fluid-example/bubblebench-common";
-import { brand, EditableField, FieldKey } from "@fluid-internal/tree";
-import { Bubble } from "./bubble";
-import { ClientTreeProxy, BubbleTreeProxy } from "./schema";
+import { EditableField } from "@fluid-internal/tree";
+import { BubbleWrapper } from "./bubble";
+import { Client, FlexBubble } from "./schema";
```

Comment:
BTW - 'IBubble' might be a good naming convention for the flex types, if Bubblebench hadn't already defined it.

Reply:
I don't think so. It's entirely unsuited for reading data.

Copied from intelisense:
```typescript
type FlexBubble = {
    [typeNameSymbol]?: "BubbleBenchAppStateBubble-1.0.0" | undefined;
    x: number | {
        [typeNameSymbol]?: "number" | undefined;
        [valueSymbol]: number;
    };
    y: number | {
        [typeNameSymbol]?: "number" | undefined;
        [valueSymbol]: number;
    };
    r: number | {
        [typeNameSymbol]?: "number" | undefined;
        [valueSymbol]: number;
    };
    vx: number | {
        [typeNameSymbol]?: "number" | undefined;
        [valueSymbol]: number;
    };
    vy: number | {
        [typeNameSymbol]?: "number" | undefined;
        [valueSymbol]: number;
    };
}
```


The normalized form would be a better choice:
```typescript
type NormalizedBubble = {
    [typeNameSymbol]: "BubbleBenchAppStateBubble-1.0.0" & BrandedType<string, "tree.Schema">;
    x: number;
    y: number;
    r: number;
    vx: number;
    vy: number;
}
```

But really neither makes a good interface for both reading and writing the data which is why we generate multiple versions.

In general, using the same interface for both reading and writing operations (like how IBubble is used) is problematic as all changes to it are breaking changes (you can't add or remove items, like extra type information). https://github.com/microsoft/FluidFramework/pull/14421 has some general thoughts on the subject.